### PR TITLE
Replace old GA code with new GTM

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Load the GUI and check there are no js errors

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4201/
Test PASSed.

**Comment by hatched** [comment]
We'll want to add the ability to disable this from the charm in the near future.

I noticed that it's blocking the requests for the actual app js. Can you move this block to the very bottom of the body, thanks.

File: jujugui/templates/index.html.mako [hunk 1]
```diff
@@ -399,30 +400,25 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-      var s = document.getElementsByTagName('script')[0];
-      s.parentNode.insertBefore(ga, s);
-      })();
+
+
+    <!-- Google Tag Manager -->
+    <noscript><iframe src="//www.googletagmanager.com/ns.html?id=GTM-K9KCMZ"
+    height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+    <script>
+    if (window.GA_key || juju_config.GA_key) {
```
Comment (hatched) [inline jujugui/templates/index.html.mako]: Thanks for moving this to the bottom but in sandbox mode it appears that GA_key is not set so it never actually gets run.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4213/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4224/
Test PASSed.

**Comment by anthonydillon** [comment]
Updated the if statement to use the correct window variable. This will still not load in sandbox or local but the GA code is set on prod so this will only run on production.

**Comment by hatched** [comment]
Thanks for the updates! :+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4226/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4228/
Test PASSed.

**Comment by kadams54** [comment]
:+1: LGTM!

**Comment by anthonydillon** [comment]
Thanks :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**